superset/assets/src/dashboard/components/filterscope/FilterScopeSelector.jsx

```diff
@@ -0,0 +1,518 @@
+  filterImmuneSliceFields: PropTypes.object.isRequired,
+  setDirectPathToChild: PropTypes.func.isRequired,
+  onCloseModal: PropTypes.func.isRequired,
+};
+
+export default class FilterScopeSelector extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    const {
+      dashboardFilters,
+      filterImmuneSlices,
+      filterImmuneSliceFields,
+      layout,
+    } = props;
+
+    if (Object.keys(dashboardFilters).length > 0) {
+      // display filter fields in tree structure
+      const filterFieldNodes = getFilterFieldNodesTree({
+        dashboardFilters,
+        isSingleEditMode: true,
+      });
+      this.allfilterFields = [];
+      filterFieldNodes.forEach(({ children }) => {
+        children.forEach(child => {
+          this.allfilterFields.push(child.value);
+        });
+      });
+
+      if (filterFieldNodes.length && filterFieldNodes[0].children) {
+        this.defaultFilterKey = filterFieldNodes[0].children[0].value;
+      }
+      const checkedFilterFields = [this.defaultFilterKey];
+      // expand defaultFilterKey
+      const { chartId } = getChartIdAndColumnFromFilterKey(
+        this.defaultFilterKey,
+      );
+      const expandedFilterIds = [chartId];
+
+      // display checkbox tree of whole dashboard layout
+      const filterScopeMap = Object.values(dashboardFilters).reduce(
+        (map, { chartId: filterId, columns }) => {
+          const filterScopeByChartId = Object.keys(columns).reduce(
+            (mapByChartId, columnName) => {
+              const filterKey = getDashboardFilterKey({
+                chartId: filterId,
+                column: columnName,
+              });
+              const nodes = getFilterScopeNodesTree({
+                components: layout,
+                isSingleEditMode: true,
+                checkedFilterFields,
+                selectedChartId: filterId,
+              });
+              const checked = getCurrentScopeChartIds({
+                scopeComponentIds: ['ROOT_ID'], // dashboardFilters[chartId].scopes[columnName],
+                filterField: columnName,
+                filterImmuneSlices,
+                filterImmuneSliceFields,
+                components: layout,
+              });
+              const expanded = getFilterScopeParentNodes(nodes, 1);
+              return {
+                ...mapByChartId,
+                [filterKey]: {
+                  // unfiltered nodes
+                  nodes,
+                  // filtered nodes in display if searchText is not empty
+                  nodesFiltered: nodes.slice(),
+                  checked,
+                  expanded,
+                },
+              };
+            },
+            {},
+          );
+
+          return {
+            ...map,
+            ...filterScopeByChartId,
+          };
+        },
+        {},
+      );
+
+      this.state = {
+        showSelector: true,
+        activeKey: this.defaultFilterKey,
+        searchText: '',
```

Comment:
can we default this to `null` instead of empty string? That'll make some of the other checks along the way more explicit (e.g. `searchText !== null` instead of `!searchText`)

Reply:
1. If i initialize the string type a string value, i will feel confident to do string operation on it, like append, truncate, split, and don't need to check and handle null case.
2. isn't `!searchText` cooler than `searchText !== null ` 😂
3. if there has no proven evidence about performance difference, why do we bother to change it?

Reply:
I think 2) and 3) aren't really significant concerns here.

using `null` is quite common esp in TypeScript. I agree that this allows for checking explicitly if there is a value or not, since there is difference in meaning between setting something to `null` vs `empty string`

Reply:
fixed. use null as initial value

Reply:
i don't see the change in this PR. Did you forget to push?